ipfx/aibs_data_set.py

```diff
@@ -70,101 +68,35 @@ def extract_sweep_meta_data(self):
                     "Unable to determine clamp mode in " + sweep_name)
 
             # bridge balance
-            bridge_balance = notebook.get_value(
+            bridge_balance = self.notebook.get_value(
                 "Bridge Bal Value", sweep_num, None)
             sweep_record["bridge_balance_mohm"] = bridge_balance
 
             # leak_pa (bias current)
-            bias_current = notebook.get_value(
+            bias_current = self.notebook.get_value(
                 "I-Clamp Holding Level", sweep_num, None)
             sweep_record["leak_pa"] = bias_current
 
             # ephys stim info
-            scale_factor = notebook.get_value("Scale Factor", sweep_num, None)
+            scale_factor = self.notebook.get_value("Scale Factor", sweep_num, None)
             if scale_factor is None:
                 raise Exception(
                     "Unable to read scale factor for " + sweep_name)
 
             sweep_record["stimulus_scale_factor"] = scale_factor
 
             # PBS-229 change stim name by appending set_sweep_count
-            cnt = notebook.get_value("Set Sweep Count", sweep_num, 0)
+            cnt = self.notebook.get_value("Set Sweep Count", sweep_num, 0)
             stim_code_ext = stim_code + "[%d]" % int(cnt)
 
             sweep_record["stimulus_code_ext"] = stim_code_ext
             sweep_record["stimulus_code"] = stim_code
 
-            if self.ontology:
-                # make sure we can find all of our stimuli in the ontology
-                stim = self.ontology.find_one(stim_code, tag_type='code')
-                sweep_record["stimulus_name"] = stim.tags(tag_type='name')[0][-1]
-
+            self.handle_ontology(sweep_record)
```

Comment:
Here we set stimulus_name and "handle_ontology" function obscures the task. Perhaps something like this would be more explicit:
sweep_record["stimulus_name"] = get_stimulus_name(stimulus_code)

Reply:
Good point. Will change.

Reply:
@sgratiy This will change the logic though as with your proposal `sweep_record["stimulus"name"]` will exist and hold None without ontology where as the entry did not exist before. I guess we can live with that change though or?

Reply:
how about this?:
if self.ontology:
      sweep_record["stimulus_name"] = get_stimulus_name(stimulus_code)

Reply:
Done.